### ISSUE TITLE
Special method `__class_getitem__` should be a class method

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1162]
**Describe the bug**
The special method `__class_getitem__` is a class method by default and its first argument should be the class, not an instance of the class.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That the type of `cls` would be `type[Foo]`.

**Screenshots or Code**
```python
class Foo:
    def __class_getitem__(cls, item: object):
        reveal_type(cls)  # Type of "cls" is "Foo"
```

**VS Code extension or command-line**
Either, 1.1.85

**Comment by erictraut** [comment]
This is now addressed in 1.1.86, which I just published. It will also be included in the next release of pylance.
